### PR TITLE
Add instructions for Java app servers

### LINKED ISSUE #3154: Docs for instrumenting various App Servers
In the Java SIG today, @trask suggested it would be nice if we had docs for adding auto instrumentation for the various popular Java App servers.  Splunk has such a doc on their site:
https://docs.splunk.com/Observability/gdi/get-data-in/application/java/instrumentation/java-servers-instructions.html

Perhaps Splunk could contribute such content to the OTel doc site.

cc @breedx-splk

### PR DESCRIPTION
**Pull request by theletterf**
Fixes #3154

---

**Preview**: https://deploy-preview-3459--opentelemetry.netlify.app/docs/instrumentation/java/automatic/server-config/

### CODE REVIEW

**Comment by theletterf** [comment]
+CC @breedx-splk

**Comment by theletterf** [comment]
We've to check some things first, so turning this into draft for now. Please do not merge until ready.

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: Configuring application servers
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:2]: How about:

```suggestion
title: Application server configuration
linkTitle: App server config
```

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+---
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:1]: Maybe the file should be renamed `server-config.md`?

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+To define the path to the Java agent, add the `javaagent` argument at the end of
+the `standalone.conf` or the `standalone.conf.bat` configuration files:
+
+```shell
+# Linux / standalone.conf
+JAVA_OPTS="$JAVA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
+
+# Windows / standalone.conf.bat
+set "JAVA_OPTS=%JAVA_OPTS% -javaagent:<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:23]: You'll want to use tabs for a better UX (in particular because tab choice is persisted):

~~~suggestion
{{< tabpane text=true >}} {{% tab header="Linux (standalone.conf)" lang=Linux %}}

```sh
JAVA_OPTS="$JAVA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
```

{{% /tab %}} {{% tab header="Windows (standalone.conf.bat)" lang=Windows %}}

```bat
set "JAVA_OPTS=%JAVA_OPTS% -javaagent:<Drive>:\path\to\opentelemetry-javaagent.jar"
```

{{% /tab %}} {{< /tabpane >}}
~~~

This will render as:

> image

**NOTE**: you won't need the `tab` attributes `header` and `lang` for _other_ tabbed panes. See below for a simplified example.

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+---
+
+When instrumenting an application running on an application server, you've to
+define the path to the OpenTelemetry Java agent. The way to define the path
+differs from server to server.
+
+## JBoss EAP / WildFly
+
+To define the path to the Java agent, add the `javaagent` argument at the end of
+the `standalone.conf` or the `standalone.conf.bat` configuration files:
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:15]:
```suggestion
the standalone configuration file:
```

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+\<jetty_home\>/bin/jetty.sh file:
+
+```shell
+JAVA_OPTIONS="${JAVA_OPTIONS} -javaagent:/path/to/opentelemetry-javaagent.jar"
+```
+
+If you use the start.ini file to define JVM arguments, add the `javaagent`
+argument after the `--exec` option:
+
+```shell
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:43]:
```suggestion
```ini
```

File: content/en/docs/instrumentation/java/automatic/servers.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+Add the path to the Java agent using the `asadmin` tool:
+
+```shell
+# Linux
+<server_install_dir>/bin/asadmin create-jvm-options "-javaagent\:/path/to/opentelemetry-javaagent.jar"
+
+# Windows
+<server_install_dir>\bin\asadmin.bat create-jvm-options '-javaagent\:<Drive>\:\\path\\to\\opentelemetry-javaagent.jar'
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/servers.md:61]:
~~~suggestion
{{< tabpane text=true >}} {{% tab Linux %}}

```sh
<server_install_dir>/bin/asadmin create-jvm-options "-javaagent\:/path/to/opentelemetry-javaagent.jar"
```

{{% /tab %}} {{% tab Windows %}}

```powershell
<server_install_dir>\bin\asadmin.bat create-jvm-options '-javaagent\:<Drive>\:\\path\\to\\opentelemetry-javaagent.jar'
```

{{% /tab %}} {{< /tabpane >}}
~~~

**NOTE** the use of the `powershell` lang for Windows.

**Review by chalin — COMMENTED** [review]
Only a few comments since you've marked this as WIP.

Before you worked further on this, I wanted to suggest switching to the use of **tabbed panes**, as illustrated below. This will provide a better UX given that tab choice is persisted.

**Comment by chalin** [comment]
> Solves #3154

FYI, it's better to write either "Fixes X" or "Closes X" so that GH can automatically close issue X when the PR is merged.

File: content/en/docs/instrumentation/java/automatic/servers.md (hunk 1 above)
Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/servers.md:2]: I like it! Always in for less gerund :)

File: content/en/docs/instrumentation/java/automatic/servers.md (hunk 2 above)
Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/servers.md:1]: +1

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
+---
+title: Application server configuration
+linkTitle: App server config
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:3]: Can we shorten this a bit more (given that this page is quite nested in the left nav)?

```suggestion
linkTitle: Server config
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+When instrumenting an application running on an application server, you've to
+define the path to the OpenTelemetry Java agent. The way to define the path
+differs from server to server.
+
+## JBoss EAP / WildFly
+
+To define the path to the Java agent, add the `javaagent` argument at the end of
+the standalone configuration file:
+
+{{< tabpane text=true >}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:18]:
```suggestion
{{< tabpane text=true persist=lang >}}

```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+Add the path to the Java agent to your startup script:
+
+```shell
+# Linux: Add the following line to \<tomcat_home\>/bin/setenv.sh
+CATALINA_OPTS="$CATALINA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
+
+# Windows: Add the following line to \<tomcat_home\>binsetenv.bat
+set CATALINA_OPTS=%CATALINA_OPTS% -javaagent:"<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:104]: You might be working on this already, but I thought that I'd mention it just in case: update to use tabs here.

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+Add the path to the Java agent to your domain startup script:
+
+```shell
+# Linux: Add the following line to \<domain_home\>/bin/startWebLogic.sh
+export JAVA_OPTIONS="$JAVA_OPTIONS -javaagent:/path/to/opentelemetry-javaagent.jar"
+
+# Linux: Add the following line to \<domain_home\>binstartWebLogic.cmd
+set JAVA_OPTIONS=%JAVA_OPTIONS% -javaagent:"<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:116]: Update to use tabs here.

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+4.  Select **Java Virtual Machine**.
+
+5.  In **Generic JVM arguments**, enter the path to the agent:
+
+    ```bash
+    -javaagent:/path/to/opentelemetry-javaagent.jar
+    ```
+
+6.  Save the configuration and restart the server.
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:151]: Ensure that the rendered list is compact by removing blank lines between entries:

```suggestion
1.  Navigate to **Servers > Server type > WebSphere application servers**.
2.  Select the server.
3.  Go to **Java and Process Management > Process Definition**.
4.  Select **Java Virtual Machine**.
5.  In **Generic JVM arguments**, enter the path to the agent:
    ```bash
    -javaagent:/path/to/opentelemetry-javaagent.jar
    ```
6.  Save the configuration and restart the server.
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+2.  Go to **Configurations > server-config > JVM Settings**.
+
+3.  Select **JVM Options > Add JVM Option**.
+
+4.  Enter the path to the agent:
+
+    `-javaagent:/path/to/opentelemetry-javaagent.jar`
+
+5.  **Save** and restart the server.
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:89]: Ensure that the rendered list is compact by removing blank lines between entries:

```suggestion
1.  Open the GlassFish Admin Console at <http://localhost:4848>.
2.  Go to **Configurations > server-config > JVM Settings**.
3.  Select **JVM Options > Add JVM Option**.
4.  Enter the path to the agent:
    `-javaagent:/path/to/opentelemetry-javaagent.jar`
5.  **Save** and restart the server.
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+## WebSphere Liberty Profile
+
+Add the path to the Java agent to the `jvm.options` file:
+
+```shell
+# Single server: Edit ${server.config.dir}/jvm.options
+# All servers: Edit ${wlp.install.dir}/etc/jvm.options
+-javaagent:/path/to/opentelemetry-javaagent.jar
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:129]:
~~~suggestion
Add the path to the Java agent to the `jvm.options` file. For a single server,
edit `${server.config.dir}/jvm.options`, and for all servers, edit
`${wlp.install.dir}/etc/jvm.options`:

```ini
-javaagent:/path/to/opentelemetry-javaagent.jar
```
~~~

**Review by chalin — COMMENTED** [review]
Another pass. See inline comments.

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+differs from server to server.
+
+## JBoss EAP / WildFly
+
+To define the path to the Java agent, add the `javaagent` argument at the end of
+the standalone configuration file:
+
+{{< tabpane text=true persist=lang >}}
+
+{{% tab header="Linux (standalone.conf)" lang=Linux %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:20]:
```suggestion
{{% tab header="Linux (`standalone.conf`)" lang=Linux %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+
+```sh
+JAVA_OPTS="$JAVA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}} {{% tab header="Windows (standalone.conf.bat)" lang=Windows %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:26]:
```suggestion
{{% /tab %}} {{% tab header="Windows (`standalone.conf.bat`)" lang=Windows %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+Make sure that the domain.xml file in your domain directory contains a
+`<jmv-options>` entry for the agent.
+
+## Tomcat / TomEE
+
+Add the path to the Java agent to your startup script:
+
+{{< tabpane text=true persist=lang >}}
+
+{{% tab header="Linux (\<tomcat_home\>/bin/setenv.sh)" lang=Linux %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:96]:
```suggestion
{{% tab header="Linux (`<tomcat_home>/bin/setenv.sh`)" lang=Linux %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+
+```sh
+CATALINA_OPTS="$CATALINA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}}
+{{% tab header="Windows (\<tomcat_home\>binsetenv.bat)" lang=Windows %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:103]:
```suggestion
{{% tab header="Windows (`<tomcat_home>\bin\setenv.bat`)" lang=Windows %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+
+{{% /tab %}} {{< /tabpane >}}
+
+## WebLogic
+
+Add the path to the Java agent to your domain startup script:
+
+{{< tabpane text=true persist=lang >}}
+
+{{% tab header="Linux (\<domain_home\>/bin/startWebLogic.sh)" lang=Linux %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:117]:
```suggestion
{{% tab header="Linux (`<domain_home>/bin/startWebLogic.sh`)" lang=Linux %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+
+```sh
+export JAVA_OPTIONS="$JAVA_OPTIONS -javaagent:/path/to/opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}}
+{{% tab header="Windows (\<domain_home\>binstartWebLogic.cmd)" lang=Windows %}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:124]:
```suggestion
{{% tab header="Windows (`<domain_home>\bin\startWebLogic.cmd`)" lang=Windows %}}
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+2.  Select the server.
+3.  Go to **Java and Process Management > Process Definition**.
+4.  Select **Java Virtual Machine**.
+5.  In **Generic JVM arguments**, enter the path to the agent:
+
+    ```bash
+    -javaagent:/path/to/opentelemetry-javaagent.jar
+    ```
+
+6.  Save the configuration and restart the server.
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:161]:
```suggestion
5.  In **Generic JVM arguments**, enter the path to the agent:
    ```bash
    -javaagent:/path/to/opentelemetry-javaagent.jar
    ```
6.  Save the configuration and restart the server.
```

**Review by chalin — COMMENTED** [review]
See inline comments for improved paths in the tab headers.

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 20 above)
Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/server-config.md]: LOL, ok. Let me check that out.

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 21]
```diff
@@ -0,0 +1,159 @@
+---
+title: Application server configuration
+linkTitle: Server config
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/server-config.md:3]:
```suggestion
linkTitle: Application server
```

For me "application server" is something different to just "server", for a Java developer this might be less confusing this way. I would even be OK with `Application server config` (or `App server config`)

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+description: Learn how to define agent paths for Java application servers
+weight: 15
+cSpell:ignore: asadmin binsetenv binstart Glassfish Payara setenv
+---
+
+When instrumenting an application running on an application server, you've to
```
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/server-config.md:9]:
```suggestion
When instrumenting an app running on an Java application server, you've to
```

**Review by svrnm — CHANGES_REQUESTED** [review]

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 21 above)
Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/server-config.md:3]: Remember that the linkTitle is only used in the left nav. When someone visits the page, they see the full title.

That being said, if you still feel that `Server config` isn't an appropriate title, among the suggestions you make, I'd vote for `App server config`.

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+-javaagent:/path/to/opentelemetry-javaagent.jar
+```
+
+Restart the server after saving the file.
+
+## WebSphere Traditional
+
+Open the WebSphere Admin Console and follow these steps:
+
+1.  Navigate to **Servers > Server type > WebSphere application servers**.
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:151]: Let's quiet the markdownlinter in this specific case:

```suggestion

<!-- markdownlint-disable blanks-around-fences -->

1.  Navigate to **Servers > Server type > WebSphere application servers**.
```

Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/server-config.md:151]: Of course, the other option is to avoid the code block in the list.

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 21 above)
Comment (svrnm) [reply content/en/docs/instrumentation/java/automatic/server-config.md:3]: Then let's go for `App server config` 👍

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+title: Application server configuration
+linkTitle: Server config
+description: Learn how to define agent paths for Java application servers
+weight: 15
+cSpell:ignore: asadmin binsetenv binstart Glassfish Payara setenv
+---
+
+When instrumenting an application running on an application server, you've to
+define the path to the OpenTelemetry Java agent. The way to define the path
+differs from server to server.
```
Comment (tylerbenson) [inline content/en/docs/instrumentation/java/automatic/server-config.md:11]:
```suggestion
When instrumenting an application running on an application server with a Java Agent, you must
add the `javaagent` path to the OpenTelemetry Java agent to the JVM arguments. The way to do this
differs from server to server.
```

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 25]
```diff
@@ -0,0 +1,159 @@
...
+
+## JBoss EAP / WildFly
```
Comment (tylerbenson) [inline content/en/docs/instrumentation/java/automatic/server-config.md:13]: I think it would be nice to start with the generic option: `JAVA_TOOLS_OPTIONS`
https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html

This should help with many situations that aren't explicitly defined here.

**Review by tylerbenson — APPROVED** [review]
I really appreciate the effort you've put into migrating this over. This will be a great and valuable contribution for the community.

Thanks!

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 26]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+{{% /tab %}}
+{{% tab header="Windows (`<domain_home>\bin\startWebLogic.cmd`)" lang=Windows %}}
+
+```bat
+set JAVA_OPTIONS=%JAVA_OPTIONS% -javaagent:"<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}} {{< /tabpane >}}
```
Comment (cartermp) [inline content/en/docs/instrumentation/java/automatic/server-config.md:130]: Unfortunately, this renders poorly in the live site. Is there a way to shorten the text in the tabs?

image

**Review by cartermp — DISMISSED** [review]

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 27]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+{{% /tab %}}
+{{% tab header="Windows (`<tomcat_home>\bin\setenv.bat`)" lang=Windows %}}
+
+```bat
+set CATALINA_OPTS=%CATALINA_OPTS% -javaagent:"<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}} {{< /tabpane >}}
```
Comment (chalin) [inline content/en/docs/instrumentation/java/automatic/server-config.md:109]: Here's an example of how to put the file path in the content, and 

~~~suggestion
{{< tabpane text=true langEqualsHeader=true >}} {{% tab header=Linux %}}

```sh
# <tomcat_home>/bin/setenv.sh
CATALINA_OPTS="$CATALINA_OPTS -javaagent:/path/to/opentelemetry-javaagent.jar"
```

{{% /tab %}} {{% tab header=Windows %}}

```bat
:: <tomcat_home>\bin\setenv.bat
set CATALINA_OPTS=%CATALINA_OPTS% -javaagent:"<Drive>:\path\to\opentelemetry-javaagent.jar"
```

{{% /tab %}} {{< /tabpane >}}
~~~

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 26 above)
Comment (chalin) [reply content/en/docs/instrumentation/java/automatic/server-config.md:130]: Yeah, when the tab name is too wide, this isn't optimal on mobile. I think that we do in other pages is add the path in the tab content.

Here's an example of how to fix this: https://github.com/open-telemetry/opentelemetry.io/pull/3459/files#r1383733272

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 25 above)
Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md:13]: Added!

File: content/en/docs/instrumentation/java/automatic/server-config.md (hunk 27 above)
Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md:109]: I went for re-adding the paths as comment in the code snippet, which I think is the most visible way.

**Review by svrnm — APPROVED** [review]

**Comment by cartermp** [comment]
Sounds good. We can keep it unmerged until it's ready 🙂

**Comment by breedx-splk** [comment]
@theletterf Where is this now? 🙏🏻

**Comment by theletterf** [comment]
Still discussing with Legal... Will let you know! Thanks for the patience!

**Comment by svrnm** [comment]
do we have any updates here?

**Comment by theletterf** [comment]
Got the OK to publish this, at last! :)

**Comment by svrnm** [comment]
@open-telemetry/java-approvers PTAL!

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 28]
```diff
@@ -0,0 +1,182 @@
...
+export JAVA_TOOL_OPTIONS="-javaagent:/path/to/opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}} {{% tab header="Windows" lang=Windows %}}
+
+```bat
+set JAVA_TOOL_OPTIONS="-javaagent:<Drive>:\path\to\opentelemetry-javaagent.jar"
+```
+
+{{% /tab %}} {{< /tabpane >}}
```
Comment (laurit) [inline content/en/docs/instrumentation/java/automatic/server-config.md:32]: See https://github.com/open-telemetry/opentelemetry.io/pull/3459/files#r1396086080
Using `JAVA_TOOL_OPTIONS` can have unintended side effects. I think it is better to use the `JAVA_OPTS` way that follows. If needed using `JAVA_TOOL_OPTIONS` can be in a separate section (that is not related to wildfly)

Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md:32]: @laurit Would this do?

```suggestion
You can define the path to the Java agent by adding the `javaagent` argument to
the `JAVA_OPTS` environment variable. For example:

{{< tabpane text=true persist=lang >}}

{{% tab header="Linux" lang=Linux %}}

```sh
export JAVA_OPTS="-javaagent:/path/to/opentelemetry-javaagent.jar"
```

{{% /tab %}} {{% tab header="Windows" lang=Windows %}}

```bat
set JAVA_OPTS="-javaagent:<Drive>:\path\to\opentelemetry-javaagent.jar"
```

{{% /tab %}} {{< /tabpane >}}
```

Comment (laurit) [reply content/en/docs/instrumentation/java/automatic/server-config.md:32]: Unfortunately no. The instructions ask user to add this at the end of the `standalone.conf` because in https://github.com/mojavelinux/jboss-as/blob/1eeb03a3fa0e795980a6ed96473df3ad0115902b/build/src/main/resources/bin/standalone.conf#L40 there is a test to check whether `JAVA_OPTS` is set, if not it adds some default arguments. When you set `JAVA_OPTS` outside the script setting the default options will be skipped. Unless you also want to explain that this is going to happen I'd suggest not including this in the documentation.

Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md]: I've updated the PR to just leave the standalone method. PTAL

File: content/en/docs/instrumentation/java/automatic/server-config.md [hunk 29]
```diff
@@ -0,0 +1,163 @@
+---
+title: Application server configuration
```
Comment (tylerbenson) [inline content/en/docs/instrumentation/java/automatic/server-config.md:2]: This currently shows at the top level (in the demo version).  How do we make sure it shows up in `instrumentation/java/automatic/`?  Maybe this is a bug with the demo?

Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md:2]: Oh, I think the files moved to /languages, could it be? Checking.

Comment (theletterf) [reply content/en/docs/instrumentation/java/automatic/server-config.md]: @tylerbenson I've moved the files, check the demo: https://deploy-preview-3459--opentelemetry.netlify.app/docs/languages/java/automatic/server-config/

Comment (tylerbenson) [reply content/en/docs/instrumentation/java/automatic/server-config.md]: looks much better, thanks!

**Comment by theletterf** [comment]
@open-telemetry/docs-maintainers I think we're good to go.

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
Yay, finally!

**Comment by svrnm** [comment]
🎉 great to see this being merged :-)

**Comment by tylerbenson** [comment]
Thanks for the work on this.  Nice addition to the site!

**Comment by trask** [comment]
thanks @theletterf (and Splunk legal)!
